featuretools/primitives/standard/aggregation/num_unique.py

```diff
@@ -60,6 +60,6 @@ def finalize(s):
 
             return dd.Aggregation(self.name, chunk=chunk, agg=agg, finalize=finalize)
 
-        if self.use_string_for_pd_calc:
+        if self.use_string_for_pd_calc or agg_type == Library.SPARK:
```

Comment:
realized that we had the potential for a bug here with spark dataframes if for some reason a user set `use_string_for_pd_calc` to false but used spark dataframes. We need to use the string for spark dataframes.